README.md

```diff
@@ -2,27 +2,24 @@
 
 Course materials for Research Software Engineering course.
 
-Current plan for structure (as advertised):
+Current plan for structure:
 
 Research Software Engineering
 
-21st, 28th Feb and 7th, 14th, 21st March 2014             2:00pm - 5:00pm
-
 Content:
 In this course, you will move beyond programming, to learn how to construct reliable, readable, efficient research software in a collaborative environment. The emphasis is on practical techniques, tips, and technologies to effectively build and maintain complex code. This is a short (15 hours over 5 half-days), intensive, practical course. The content of each of the 5 half-day units is as follows:
 
-1. Code management. Collaborating around code. Distributed version control. Git. Github. Issue tracking. Code review and pull requests. Branches and merging. Software licenses. Citing software. Software sustainability.
-1. Testing scientific software. Unit testing, regression testing. Test driven design. Expectations and assertions. Mocking. Build-and-test servers. Negative testing. Sensible error messages. Managed logging. Debugging and debuggers. Coverage measurement.
+1. Code management. Collaborating around code. Distributed version control. Git. Github. Issue tracking. Code review and pull requests. Branches and merging. Continuous Integration. Citing software. Software sustainability.
+1. Testing scientific software. Sensible error messages. Error handling. Print-debugging. Debuggers. Test pyramid: unit testing, regression testing. Test driven design. Expectations and assertions. Mocking. Negative testing.
 1. Best practice in construction. Comments. Coding conventions. Documentation. Refactoring. IDEs. Configuration files. Using libraries.
 1. Design and development. Object oriented design. Software as engineering. Pragmatic use of diagram languages. Requirements engineering. Agile and Waterfall. Functional and architectural design.
```

Comment:
what are your experiences using diagram languages? I've not used them since undergrad as I never thought they would be useful, but I could be wrong here.

Reply:
In my most recent team we went through a few iterations, starting with "none", then trying a non-semantic diagramming tool (LucidChart), then settled on using graphviz and plantuml. I think there's value in having diagrams that are better maintained/versioned than whiteboard sketches, and negative value in using those fancy round-trip UML CASE tools.